src/twisted/internet/cfreactor.py

```diff
@@ -385,6 +385,13 @@ def mainLoop(self):
         finally:
             self._inCFLoop = False
 
+        assert not self.running, (
```

Comment:
a comment might help us understand why we need this assertion here... or maybe have a different assertion message that can serve to document this code.